src/node_shell_read.js

```diff
@@ -4,18 +4,9 @@
  * SPDX-License-Identifier: MIT
  */
 
-var fs;
-var nodePath;
-
-var requireNodeFS = () => {
-  // Use nodePath as the indicator for these not being initialized,
-  // since in some environments a global fs may have already been
-  // created.
-  if (!nodePath) {
-    fs = require('fs');
-    nodePath = require('path');
-  }
-};
```

Comment:
How about trying to land the "import on demand" to "always import" as a separate change?

Reply:
I just opened PR https://github.com/emscripten-core/emscripten/pull/17851 for this. I'll rebase this PR once that lands, otherwise it will cause failures in the `core2.test_dlfcn*` test suite (it tries to reference `nodePath` before it's `require()`'d somehow).